File: web/tests/reference/heaps/find-median-from-data-stream.js
class MedianFinder {
  constructor() {
    this.values = [];
  }

  addNum(num) {
    this.values.push(num);
    this.values.sort((a, b) => a - b);
  }

  findMedian() {
    const n = this.values.length;
    const mid = Math.floor(n / 2);
    if (n % 2 === 1) return this.values[mid];
    return (this.values[mid - 1] + this.values[mid]) / 2;
  }
}

File: web/tests/reference/heaps/find-median-from-data-stream.py
import bisect

class MedianFinder:
    def __init__(self):
        self.values = []

    def add_num(self, num):
        bisect.insort(self.values, num)

    def find_median(self):
        n = len(self.values)
        mid = n // 2
        if n % 2 == 1:
            return self.values[mid]
        return (self.values[mid - 1] + self.values[mid]) / 2
